packaging/tags.py

```diff
@@ -162,20 +172,55 @@ def _cpython_abis(py_version, warn=False):
     return abis
 
 
-def _cpython_tags(py_version, interpreter, abis, platforms):
-    # type: (PythonVersion, str, Iterable[str], Iterable[str]) -> Iterator[Tag]
+def cpython_tags(
+    python_version=sys.version_info[:2],  # type: PythonVersion
```

Comment:
Sure, but can I ask why? This avoids that `if` check and I have gotten pushback previously in this repo for having too many explicit conditional branches. Plus tuples are immutable so there's also no worry about accidentally mutating the constant in-place. Is it for consistency reasons?

Reply:
And does that mean you prefer an `if ... is None` check instead to tightly check for `None` versus any false value?

Reply:
I :+1:d because it makes the fact that it defaults to `sys.version_info` testable vs the current construction.

Reply:
Yea, as @chrahunt points out, my main thought was testability of the fact that we're using sys.version_info for this default (possibly via mocking out the dependency to confirm that).

That said, I don't feel strongly about this since that's the only reason it's useful in that one case and IIUC, the benefit of doing so isn't much either.

Reply:
I'm trying to finish up the integration against this branch now. Given that context, I think it is easier to use if defaulted to `None` here. For all other parameters we can do something like:

```
abis = None
if abi_from_user:
    abis = [abi_from_user]

cpython_tags(
    ...,
    abis=abis,
    ...
)
```

but for this one we'd need to either:

1. explicitly default to `python_version = sys.version_info[:2]` outside the function and have a test that we defaulted correctly, or
2. create a dictionary `args = {}` and conditionally do `args["python_version"] = python_version_from_user` and invoke this function like `cpython_tags(**args)`, which I don't think is friendly with type checkers without additional work to define the type of `args`

I would make the same argument for having `python_version=None` in `compatible_tags` too.